python/ml4ir/base/pipeline.py

```diff
@@ -413,13 +394,15 @@ def run(self):
                 )
 
             # Finish
-            self.finish(job_status, job_info)
+            # self.finish(job_status, job_info)
```

Comment:
I am planning to remove this call to finish too, if you are OK @lastmansleeping. In the same function, we call `finish` again lower so I did not find the purpose of doing it several times and overwriting files.